src/crimson/os/seastore/random_block_manager/nvme_block_device.cc

```diff
@@ -192,43 +192,50 @@ write_ertr::future<> NVMeBlockDevice::writev(
     supported_stream = WRITE_LIFE_NOT_SET;
   }
   if (is_end_to_end_data_protection()) {
-    return seastar::do_with(
-      std::move(bl),
-      [this, offset] (auto &bl) {
-      return nvme_write(offset, bl.length(), bl.c_str());
-    });
+    co_await nvme_write(offset, bl.length(), bl.c_str());
+    co_return;
   }
   bl.rebuild_aligned(super.block_size);
+  auto iovs = bl.prepare_iovs();
+  std::vector<seastar::future<>> tasks;
```

Comment:
iiuc the idea here is to create tasks in order for us to be able to wait for the futures to run in parallel.
How about `seastar::coroutine::parallel_for_each`?